### ISSUE TITLE
Can not put `media` helper into nest$

### ISSUE THREAD

**Issue by whitecolor** [typestyle/typestyle#196]
I think this should be possible:

![image](https://user-images.githubusercontent.com/736697/31317037-07e663fc-ac53-11e7-9a5a-86b83690c120.png)

```ts
$nest: {
     media ({ 
       maxWidth: '500px'
  }, { }),
```

Why not? Or I don't get the design intention behind `media` helper.

**Comment by notoriousb1t** [comment]
I understand why it would look like that should work, but it simply isn't possible in JavaScript.   

The style function can take multiple definitions, so the intended usage of media is:

```ts
const blueWhenSmall = style(
  {
    color: 'red'
  },
  media({ maxWidth: 960 }, {
    color: 'blue'
  })
);

<h2 className={blueWhenSmall}>
  Resize to less than 960px make this blue
</h2>
```

Demo:
https://typestyle.github.io/play/#src=const%20blueWhenSmall%20=%20style(%0A%20%20%7B%0A%20%20%20%20color:%20'red'%0A%20%20%7D,%0A%20%20media(%7B%20maxWidth:%20960%20%7D,%20%7B%0A%20%20%20%20color:%20'blue'%0A%20%20%7D)%0A);%0A%0A%3Ch2%20className=%7BblueWhenSmall%7D%3E%0A%20%20Resize%20to%20less%20than%20960px%20make%20this%20blue%0A%3C/h2%3E

**Comment by whitecolor** [comment]
> I understand why it would look like that should work, but it simply isn't possible in JavaScript.

Sory I just forget to spread.

From lib code:

```ts
  const object  = {
    $nest: {
      [stringMediaQuery]: extend(...objects)
    }
  } as types.CSSProperties;
  return object;
```
First probably not correct imperative type `types.CSSProperties` should be `types.NestedCSSProperties`

Then this thing would work

```
$nest: {
     ...media ({ 
       maxWidth: '500px'
  }, { }).$nest,
```

Do I manage to explan what I mean?

**Comment by notoriousb1t** [comment]
I merged the PR, nice catch there.   I believe that code should work now.

**Comment by notoriousb1t** [comment]
Released in 1.4.2
